### PR TITLE
Reviewed and fixed files according to golang-ci linters

### PR DESCRIPTION
**Pull request by Raviv-S**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    - path: _test\.go
+      linters:
+       - dupl
+       - revive
+    - path: pkg/test/
+      linters:
+       - revive
+       - gomnd
+       - gocritic
+    - linters:
```
Comment (roytman) [inline .golangci.yml:134]: what is the path of this linter

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      - lll
+      source: "^// "
+    - linters:
```
Comment (roytman) [inline .golangci.yml]: the same

File: manager/apis/motion/v1alpha1/batchtransfer_webhook.go [hunk 3]
```diff
@@ -24,9 +24,17 @@ import (
...
-func (r *BatchTransfer) SetupWebhookWithManager(mgr ctrl.Manager) error {
+const (
+	DefaultFailedJobHistoryLimit     = 5
+	MaxFailedJobHistoryLimit         = 20
+	DefaultSuccessfulJobHistoryLimit = 5
+	MaxSuccessfulJobHistoryLimit     = 20
+	batchTransferImage               = "ghcr.io/fybrik/mover:latest"
+)
+
+func (batchTransfer *BatchTransfer) SetupWebhookWithManager(mgr ctrl.Manager) error {
```
Comment (roytman) [inline manager/apis/motion/v1alpha1/batchtransfer_webhook.go:35]: why did you rename `r` to batchTransfer is it was asked by one of the linters ?

File: manager/apis/motion/v1alpha1/batchtransfer_webhook.go [hunk 4]
```diff
@@ -243,7 +250,7 @@ func validateDataStore(path *field.Path, store *DataStore) []*field.Error {
 		}
 
 		// Validate Kafka schema registry url
-		if len(store.Kafka.SchemaRegistryURL) != 0 {
+		if store.Kafka.SchemaRegistryURL != "" {
```
Comment (roytman) [inline manager/apis/motion/v1alpha1/batchtransfer_webhook.go:253]: which lint asked this replacement ?

File: manager/controllers/utils/utils.go [hunk 5]
```diff
@@ -11,6 +11,13 @@ import (
 	"runtime"
 )
 
+const (
+	StepNameHashLength       = 10
```
Comment (roytman) [inline manager/controllers/utils/utils.go]: all constants for the package internal usage -> low case

**Review by roytman — COMMENTED** [review]
several small comments, 
+ where is ` .pre-commit-config.yaml`

File: manager/apis/motion/v1alpha1/batchtransfer_webhook.go (hunk 3 above)
Comment (Raviv-S) [reply manager/apis/motion/v1alpha1/batchtransfer_webhook.go:35]: Yes, "stylecheck" raised it.
`ST1016: methods on the same type should have the same receiver name`

File: manager/apis/motion/v1alpha1/batchtransfer_webhook.go (hunk 4 above)
Comment (Raviv-S) [reply manager/apis/motion/v1alpha1/batchtransfer_webhook.go:253]: gocritic:
`emptyStringTest: replace 'len(store.Kafka.DataFormat) != 0' with 'store.Kafka.DataFormat != ""'`

File: .golangci.yml (hunk 1 above)
Comment (Raviv-S) [reply .golangci.yml:134]: A path is not necessary because I want this linter to work on the whole repo and not on a single file, it's suppose to ignore length of long comments.
The reason I added it is to ignore the long kubebuilder markers.

File: manager/controllers/utils/utils.go (hunk 5 above)
Comment (Raviv-S) [reply manager/controllers/utils/utils.go]: Missed that, I'll fix it

File: .golangci.yml (hunk 1 above)
Comment (roytman) [reply .golangci.yml:134]: >  I want this linter to work on the whole repo

I think you ignored the linter, you added it under `exclude-rule`

File: .golangci.yml (hunk 2 above)
Comment (Raviv-S) [reply .golangci.yml]: This linter is to specifically ignore this message:
`SA1019: fake.NewFakeClientWithScheme is deprecated: Please use NewClientBuilder instead. (staticcheck)`
on: manager/controllers/motion/batchtransfer_controller_unit_test.go
I can add the test file path here

File: .golangci.yml (hunk 1 above)
Comment (Raviv-S) [reply .golangci.yml:134]: That what I meant, I'll rephrase it:
I want the exception rule of this linter to work on the whole repo, meaning on all the files long comment will not raise a linter warning.

Comment (roytman) [reply .golangci.yml:134]: therefore, maybe will be better to disable this linter for now

**Review by roytman — APPROVED** [review]
